tmol/score/common/device_operations.cuda.impl.cuh

```diff
@@ -61,6 +61,16 @@ struct DeviceOperations<tmol::Device::CUDA> {
     mgpu::mem_to_shared<N_T, WIDTH>(src, threadIdx.x, n, dst, false);
   }
 
+  template <int N_T, int WIDTH, typename TD, typename TS>
+  __device__ static void copy_and_cast_contiguous_data(
```

Comment:
How about "copy_contiguous_data_and_cast"? It takes contiguous data from global memory to type TS (type of the src) and casts it to type TD (type of the dst) before assigning to dst. I currently am storing some small integers using `int32_t` and want to store them as `unsigned char`s in shared memory and do that data loading efficiently and to avoid code duplication.